Segfault with --no-dry-run

@ekristen thanks again for the help yesterday.  I have a new issue with a segfault.  Note I've omitted trace output, but there's no additional logging that happens after "scan complete..." and the panic.

config.yaml

```
regions:
    - eastus

blocklist:
    - fake-tenant-id

accounts:
    redacted-tenant-id:
        filters:
            ResourceGroup:
                - Default
                - NetworkWatcherRG
            PolicyAssignment:
                - SecurityCenterBuiltIn
```

--no-dry-run output:
```
INFO[0000] configured locations[eastus]

eastus/rg0 - ResourceGroup - delete-me_group - [Location: "eastus", Name: "delete-me_group"] - would remove
eastus/rg0 - ResourceGroup - deleteme - [Location: "eastus", Name: "deleteme"] - would remove
eastus/rg0 - VirtualNetwork - delete-me-vnet - [Name: "delete-me-vnet", ResourceGroup: "delete-me_group"] - would remove
eastus/rg0 - SSHPublicKey - somekey - [Name: "somekey", ResourceGroup: "delete-me_group"] - would remove
eastus/rg0 - NetworkSecurityGroup - delete-me-nsg - [Location: "eastus", Name: "delete-me-nsg"] - would remove
eastus/rg0 - VirtualMachine - delete-me - [Name: "delete-me", ResourceGroup: "delete-me_group"] - would remove
eastus/rg0 - NetworkInterface - delete-me293_z1 - [Name: "delete-me293_z1", ResourceGroup: "delete-me_group"] - would remove
eastus/rg0 - PublicIPAddresses - delete-me-ip - [Name: "delete-me-ip", ResourceGroup: "delete-me_group"] - would remove
eastus/rg0 - Disk - delete-me_disk1_23497234987234hjsdf - [Name: "delete-me_disk1_23497234987234hjsdf", ResourceGroup: "delete-me_group"] - would remove
eastus/rg1 - ResourceGroup - delete-me_group - [Location: "eastus", Name: "delete-me_group"] - would remove
eastus/rg1 - ResourceGroup - deleteme - [Location: "eastus", Name: "deleteme"] - would remove
eastus/rg2 - ResourceGroup - delete-me_group - [Location: "eastus", Name: "delete-me_group"] - would remove
eastus/rg2 - ResourceGroup - deleteme - [Location: "eastus", Name: "deleteme"] - would remove
Scan complete: 3123 total, 13 nukeable, 3110 filtered.

panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19cc742]

goroutine 1 [running]:
main.main.func1()
	/home/runner/work/azure-nuke/azure-nuke/main.go:25 +0x54
panic({0x1adac40?, 0x2405950?})
	/opt/hostedtoolcache/go/1.21.6/x64/src/runtime/panic.go:914 +0x21f
github.com/ekristen/azure-nuke/resources.(*ResourceGroup).Remove(0x38?, {0x1d3f880?, 0x2448840?})
	/home/runner/work/azure-nuke/azure-nuke/resources/resource-group.go:38 +0x22
github.com/ekristen/libnuke/pkg/nuke.(*Nuke).HandleRemove(0xc0001a8f50?, {0x1d3f880?, 0x2448840?}, 0xc00150e840)
	/home/runner/go/pkg/mod/github.com/ekristen/libnuke@v0.6.0/pkg/nuke/nuke.go:474 +0x2e
github.com/ekristen/libnuke/pkg/nuke.(*Nuke).HandleQueue(0xc000213400, {0x1d3f880, 0x2448840})
	/home/runner/go/pkg/mod/github.com/ekristen/libnuke@v0.6.0/pkg/nuke/nuke.go:436 +0x105
github.com/ekristen/libnuke/pkg/nuke.(*Nuke).run(0xc000213400, {0x1d3f880, 0x2448840})
	/home/runner/go/pkg/mod/github.com/ekristen/libnuke@v0.6.0/pkg/nuke/nuke.go:225 +0x85
github.com/ekristen/libnuke/pkg/nuke.(*Nuke).Run(0xc000213400, {0x1d3f880, 0x2448840})
	/home/runner/go/pkg/mod/github.com/ekristen/libnuke@v0.6.0/pkg/nuke/nuke.go:209 +0x17a
github.com/ekristen/azure-nuke/pkg/commands/nuke.execute(0xc0001dac40)
	/home/runner/work/azure-nuke/azure-nuke/pkg/commands/nuke/command.go:183 +0x1cc5
github.com/urfave/cli/v2.(*Command).Run(0xc00014a6c0, 0xc0001da600)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x583
github.com/urfave/cli/v2.(*App).RunContext(0xc00020ed00, {0x1d3f880?, 0x2448840}, {0xc000022100, 0x8, 0x8})
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0xaa5
github.com/urfave/cli/v2.(*App).Run(...)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
	/home/runner/work/azure-nuke/azure-nuke/main.go:45 +0x20d
```

--- a/.github/workflows/goreleaser.yml
+++ b/.github/workflows/goreleaser.yml
@@ -94,7 +94,7 @@ jobs:
         run: |
           docker images --format "{{.Repository}}:{{.Tag}}" | grep "${{ github.repository }}" | xargs -L1 docker push
       - name: upload artifacts
-        if: github.event.pull_request.base.ref == 'main'
+        if: ${{ github.event.pull_request.base.ref == 'main' || github.event_name == 'workflow_dispatch' }}
         uses: actions/upload-artifact@v4
         with:
           name: binaries

--- a/resources/budget.go
+++ b/resources/budget.go
@@ -29,9 +29,10 @@ func init() {
 }
 
 type Budget struct {
-	client *budgets.BudgetsClient
-	name   *string
-	id     *string
+	client         *budgets.BudgetsClient
+	name           *string
+	id             *string
+	subscriptionId *string
 }
 
 type BudgetLister struct{}
@@ -68,9 +69,10 @@ func (l BudgetLister) List(pctx context.Context, o interface{}) ([]resource.Reso
 
 	for _, entry := range *list.Model {
 		resources = append(resources, &Budget{
-			client: client,
-			name:   entry.Name,
-			id:     entry.Id,
+			client:         client,
+			name:           entry.Name,
+			id:             entry.Id,
+			subscriptionId: ptr.String(opts.SubscriptionId), // note: this is just the guid
 		})
 	}
 
@@ -80,8 +82,11 @@ func (l BudgetLister) List(pctx context.Context, o interface{}) ([]resource.Reso
 }
 
 func (r *Budget) Remove(ctx context.Context) error {
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(10*time.Second))
+	defer cancel()
+
 	_, err := r.client.Delete(ctx, budgets.ScopedBudgetId{
-		Scope:      "",
+		Scope:      fmt.Sprintf("/subscriptions/%s", ptr.ToString(r.subscriptionId)),
 		BudgetName: ptr.ToString(r.name),
 	})
 	return err

--- a/resources/resource-group.go
+++ b/resources/resource-group.go
@@ -31,7 +31,7 @@ type ResourceGroup struct {
 	client         *resourcegroups.ResourceGroupsClient
 	name           *string
 	location       string
-	subscriptionId *string
+	subscriptionId string
 }
 
 func (r *ResourceGroup) Remove(ctx context.Context) error {
@@ -44,6 +44,7 @@ func (r *ResourceGroup) Properties() types.Properties {
 
 	properties.Set("Name", r.name)
 	properties.Set("Location", r.location)
+	properties.Set("SubscriptionId", r.subscriptionId)
 
 	return properties
 }
@@ -84,9 +85,10 @@ func (l ResourceGroupLister) List(ctx context.Context, o interface{}) ([]resourc
 
 	for _, entity := range *list.Model {
 		resources = append(resources, &ResourceGroup{
-			client:   client,
-			name:     entity.Name,
-			location: entity.Location,
+			client:         client,
+			name:           entity.Name,
+			location:       entity.Location,
+			subscriptionId: opts.SubscriptionId,
 		})
 	}
 

--- a/resources/resource-group.go
+++ b/resources/resource-group.go
@@ -35,7 +35,7 @@ type ResourceGroup struct {
 }
 
 func (r *ResourceGroup) Remove(ctx context.Context) error {
-	_, err := r.client.Delete(ctx, commonids.NewResourceGroupID(*r.subscriptionId, *r.name), resourcegroups.DefaultDeleteOperationOptions())
+	_, err := r.client.Delete(ctx, commonids.NewResourceGroupID(r.subscriptionId, *r.name), resourcegroups.DefaultDeleteOperationOptions())
 	return err
 }
 

